### PR TITLE
 Dead connection handling fixes

### PR DESCRIPTION
**Pull request by andris-szatmari**
- fix resending logic: don't duplicate messages in messagestore when trying to send to a dead connection
- fix for readding confirmation select confirmation listeners and exchange, queue and binding declarations on connection lost
- reformatted long lines
- updated scalariform
- set namehashing to match kinja repos, so it could be included as dependency from filesystem

@leventem can you please review?

### CODE REVIEW

File: src/main/scala/com/kinja/amqp/UnconfirmedMessageRepeater.scala [hunk 1]
```diff
@@ -98,11 +110,15 @@ class UnconfirmedMessageRepeater(
...
 		republishTimeout: FiniteDuration
 	)(implicit ec: ExecutionContext): Unit = {
-		msgs.map { msg =>
+		msgs.foreach { msg =>
 			val result = Try(Await.result(producer.publish(msg.routingKey, Json.parse(msg.message)), republishTimeout))
 			result match {
 				case Success(_) => deleteMessageAndMatchingConfirm(msg, confs, transactional)
-				case Failure(ex) => logger.warn(s"""Couldn't resend message: $msg, ${ex.getMessage}""")
+				case Failure(ex: TimeoutException) =>
+					deleteMessageAndMatchingConfirm(msg, confs, transactional)
```
Comment (leventem) [inline src/main/scala/com/kinja/amqp/UnconfirmedMessageRepeater.scala:118]: If the publish times out, can we be sure that the new message will be saved to the msg store? What if there's a db connection problem?

**Comment by leventem** [comment]
Other than the comment above, it lgtm!

File: src/main/scala/com/kinja/amqp/UnconfirmedMessageRepeater.scala (hunk 1 above)
Comment (andris-szatmari) [inline src/main/scala/com/kinja/amqp/UnconfirmedMessageRepeater.scala:118]: I guess this way the delete itself won't work either, but it's a tricky
question.
What is sure that without this the messeges get duplicated when rabbit is
unreachable, and unfortunately we get back this exception, not a more
specific one.
I'll check out what can we do about it tomorrow.

Comment (andris-szatmari) [inline src/main/scala/com/kinja/amqp/UnconfirmedMessageRepeater.scala:118]: Okay, so as I see slick throws a `java.sql.SQLTimeoutException`, which isn't a subclass of this `java.util.concurrent.TimeoutException`, so they should encounter

**Pull request merged**